src/main/java/com/percero/agents/sync/dao/DAORegistry.java

```diff
@@ -0,0 +1,39 @@
+package com.percero.agents.sync.dao;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.percero.agents.sync.services.DAODataProvider;
+import com.percero.agents.sync.services.DataProviderManager;
+import com.percero.framework.vo.IPerceroObject;
+
+import edu.emory.mathcs.backport.java.util.Collections;
+
+public class DAORegistry {
+	
+	private static DAORegistry instance = null;
+	
+	public static DAORegistry getInstance() {
+		if (instance == null) {
+			instance = new DAORegistry();
+		}
+		return instance;
+	}
+
+	public DAORegistry() {
+		instance = this;
+		
+		DataProviderManager.getInstance().addDataProvider(DAODataProvider.getInstance()); 
+	}
+
+	@SuppressWarnings({ "unchecked" })
+	private static Map<String, IDataAccessObject<? extends IPerceroObject>> dataAccessObjects = Collections.synchronizedMap(new HashMap<String, IDataAccessObject<? extends IPerceroObject>>());
+	
+	public void registerDataAccessObject(String name, IDataAccessObject<? extends IPerceroObject> dataAccessObject) {
+		dataAccessObjects.put(name, dataAccessObject);
+	}
+	
+	public IDataAccessObject<? extends IPerceroObject> getDataAccessObject(String name) {
```

Comment:
It would be cool to use generics here. Instead of having the keys be Strings have them be classes. That way you can do the casting on the inside of the method rather than having to always cast when you access the daos.
```java
private Map<Class<? extends IPerceroObject>, IDataAccessObject<? extends IPerceroObject>> dataAccessObjects
...

public <T extends IPerceroObject> void registerDataAccessObject(IDataAccessObject<T> dao){
  dataAccessObjects.put(dao.getClass(), dao);
}
public <T extends IPerceroObject> IDataAccessObject<T> getDataAccessObject(Class<T> clazz){
  return (IDataAccessObject<T>) dataAccessObjects.get(clazz);
}